lib/puppet/provider/mongodb_database/mongodb.rb

```diff
@@ -9,12 +9,16 @@
   def self.instances
     require 'json'
 
-    pre_cmd = 'db.getMongo().setReadPref("primaryPreferred")'
-    dbs = JSON.parse mongo_eval("#{pre_cmd};EJSON.stringify(db.getMongo().getDBs())")
+    if db_ismaster
+      dbs = JSON.parse mongo_eval('EJSON.stringify(db.getMongo().getDBs())')
```

Comment:
When setting mongodb::server::replset_members mongodb::server also needs to manage the 'admin' user.
The only way to configure this is by setting mongodb::server::create_admin to true.
This then manages both the admin db and the admin user ...
When running with  mongodb::server::create_admin on a secondary server, getBSs() fail.

Reply:
Does anyone know why we manage the 'admin' db?
From what I could find the admin db is always existing when mongod is started.

Reply:
I created an issue for this #746

Reply:
Getting the DBs on a secondary definitely works on 4.4, just verified by hand.
Also, both my tests (replica set with 3 members) and the acceptance tests with replica sets pass. Getting the DB's is executed several times on a secondary.